src/content/plugins.md

```diff
@@ -0,0 +1,331 @@
+    }
+}
+```
+
+## Constructor
+
+```
+    constructor (inputNode, options) {
+        super([inputNodes], options);
+    }
+```
+
+`inputNode`: This can be one of two things, a string or another Broccoli plugin. Some plugins will require multiple
+`inputNodes` and some only require a single. Think about what your plugin is doing, does it need to operate on multiple
+input directories or a single one. The majority of plugins only require a single directory. The `broccoli-plugin`
+base class requires this argument to be an array however, so always make sure pass an array to `super()`.
+
+If a string is passed, Broccoli expects this to be a source directory within the project and automatically converts
+this into a [broccoli-source](https://github.com/broccolijs/broccoli-source) plugin.
+
+`options`: (all optional)
+
+```js
+{
+    name: '',
+    annotation: '',
+    persistentOutput: false,
+    needsCache: false,
+}
+```
+
+`options.name`: Custom name used when debugging/printing stack traces. Broccoli will use the name of your
+plugin if this field is not supplied.
+
+`options.annotation`: In addition to the plugin name, Broccoli uses annotations to provide a descriptive label used
+during debugging/printing stack traces. This is not often set by the plugin author, but by the consumer of the
+plugin to tell multiple instances of the same plugin apart.
+
+`options.persistentOutput`: By default, Broccoli will provide an empty `outputPath` for the plugin to write files to.
+This applies when running the `serve` command, where the Broccoli process is still running and `build()` is called
+on a plugin again when some input files have changed and a rebuild is triggered.
+
+If this option is true, the output directory is not automatically emptied between rebuilds. This may be useful if your
+plugin implements caching to allow the build method to be skipped. Whenever the Broccoli process exits (after `build`
+or quitting `serve`), all directories are deleted.
+
+`options.needsCache`: Despite the name, `needsCache` doesn't provide caching for the plugin. If true, a directory
+is created for the plugin to store temporary files that may be needed between rebuilds that are not included in the
+build output. Broccoli sets the path to this directory to `this.cachePath`.
+
+## Build
+
+Build can do anything you can do in node. Additionally, if `build()` returns a promise, Broccoli will wait until the
+promise resolves before continuing the rest of the build. Broccoli only builds one plugin at a time in order, from top
+to bottom of the build graph.
+The build method is where the grunt (lol, intentional pun) of the work happens.
+
+This function will typically access the following properties:
+
+`this.inputPaths`: An array of paths on disk corresponding to each node in inputNodes. Your plugin will read
+files from these paths. These paths are auto-generated by Broccoli and will typically be the output path of that a
+previous plugin has written to.
```

Comment:
One of my stumbling blocks yesterday was wondering whether inputPaths and outputPaths were stable values (as in the tmp folder they point to) or whether they would change from `build()` to `build()`. I believe they're stable (maybe I'm wrong?) but seeing this stated here would've helped me more quickly form a mental model around how the plugin "scratch" space is constructed, what's dynamic, what's static throughout the lifetime of a Broccoli process.

Reply:
Good point, added